File: qb-engineer-ui/src/app/features/invoices/models/invoice-list-item.model.ts
export interface InvoiceListItem {
  id: number;
  invoiceNumber: string;
  customerId: number;
  customerName: string;
  status: string;
  invoiceDate: Date;
  dueDate: Date;
  total: number;
  amountPaid: number;
  balanceDue: number;
  createdAt: Date;
}

File: qb-engineer-server/qb-engineer.core/Models/InvoiceListItemModel.cs
namespace QBEngineer.Core.Models;

public record InvoiceListItemModel(
    int Id,
    string InvoiceNumber,
    int CustomerId,
    string CustomerName,
    string Status,
    DateTimeOffset InvoiceDate,
    DateTimeOffset DueDate,
    decimal Total,
    decimal AmountPaid,
    decimal BalanceDue,
    DateTimeOffset CreatedAt);
